tests/utils/math_utils.py

```diff
@@ -76,6 +80,16 @@ def test_weighted_median():
     assert median == 4
 
 
+def test_weighted_median_error():
```

Comment:
can you add a test with one unit only? that should now work, right? Also please add a test at exactly 50% of the weight.